common/src/main/java/org/openmrs/analytics/HttpUtil.java

```diff
@@ -46,10 +86,35 @@ public String executeRequest(HttpUriRequest request) {
 				throw new RuntimeException();
 			}
 			return byteStream.toString();
-		}
-		catch (IOException e) {
+		} catch (IOException e) {
 			log.error("Error in opening url: " + request.getURI().toString() + " exception: " + e);
 			return "";
+		} finally {
+			httpClient.close();
+		}
+	}
+
+	public HttpURLConnection getServerConnection(String Serverurl) throws Exception {
```

Comment:
@bashir2  , i left in these util methods , thinking they might be useful at some point .
should i get rid of them ??